app/assets/stylesheets/components/forms.scss

```diff
@@ -0,0 +1,140 @@
+@import '../vars/_variables.scss';
+
+%form-styling {
+  background-color: $smoke-00;
+  border: 1px solid $smoke-30;
+  appearance: none;
+  border-radius: $br-default;
+  transition: all $transition-props;
+
+  &-hover {
+    border-color: $smoke-40;
+  }
+
+  &-disabled {
+    color: $smoke-60;
+    background-color: $smoke-10;
+  }
+
+  &-checked {
+    background-color: $dolphin;
+    border-color: $dolphin-darker;
+    color: $snowflake;
+    background-position: center center;
+    background-repeat: no-repeat;
+  }
+
+  &-checked-disabled {
+    background-color: $dolphin-lighter;
+    border-color: $dolphin-lighter;
+    color: rgba($snowflake, 0.5);
+  }
+}
+
+// Text inputs and textareas.
+.crayons-textfield {
+  line-height: $lh-base;
+  padding: 0.5em;
+  font-family: inherit;
+  font-size: $fs-base;
+  width: 100%;
+  @extend %form-styling;
+
+  &::placeholder {
```

Comment:
Double `::` is this a typo?

Reply:
There's a [tiny difference](https://stackoverflow.com/a/41867706/1147887) between `:` and `::`. Tl;dr `:` is for states (hover, active, checked, ...) and `::` is for actual elements (placeholder is actually an element INSIDE input... Technically `after` and `before` should be prefixed with `::` but they all work with both... Sooo.... 🤷‍♂️ :D 

But it's a very good catch - I have to make double check if I actually follow that rule :).

Reply:
TIL part 2!